api/response/schema.py

```diff
@@ -0,0 +1,100 @@
+import graphene
+from graphene_sqlalchemy import SQLAlchemyObjectType
+from api.response.models import Response as ResponseModel
+from utilities.utility import validate_empty_fields, validate_rating_field
+from helpers.auth.user_details import get_user_from_db
+from graphql import GraphQLError
+from api.room.schema import Room
+from api.question.models import Question
+from helpers.auth.authentication import Auth
+
+
+class Response(SQLAlchemyObjectType):
+    class Meta:
+        model = ResponseModel
+
+
+class CreateRate(graphene.Mutation):
+    class Arguments:
+        question_id = graphene.Int(required=True)
+        room_id = graphene.Int(required=True)
+        rate = graphene.Int(required=True)
+    rating = graphene.Field(Response)
+
+    # @Auth.user_roles('Default User')
+    def mutate(self, info, **kwargs):
+        query = Room.get_query(info)
+        room = query.filter_by(id=kwargs['room_id']).first()
+        if not room:
+            raise GraphQLError("Non-existent room id")
+        get_question = Question.query.filter_by(id=kwargs['question_id']).first() # noqa
+        if not get_question:
+            raise GraphQLError("Question does not exist")
+        question_type = get_question.question_type
+        if question_type.lower() == 'rate':
+            validate_rating_field(**kwargs)
+        else:
+            raise GraphQLError("Select a rating question")
+        user = get_user_from_db()
+        rating = ResponseModel(**kwargs, user_id=user.id)
+        rating.save()
+        return CreateRate(rating=rating)
+
+
+class CreateCheck(graphene.Mutation):
```

Comment:
This would be unnecessary if we had if had a question type and then in the response, restrict answers based on question.type

Reply:
I didn't create this, I will have a sync with @migot01, he created the mutation

Reply:
@shemogumbe I have implemented the requested changes